core/include/traccc/clusterization/impl/measurement_creation.ipp

```diff
@@ -17,37 +17,43 @@ inline scalar signal_cell_modelling(
     return signal_in;
 }
 
-TRACCC_HOST_DEVICE
-inline vector2 position_from_cell(
-    const cell& cell,
+template <typename T>
+TRACCC_HOST_DEVICE inline vector2 position_from_cell(
+    const edm::silicon_cell<T>& cell,
```

Comment:
And here, this templating is super confusing.

Reply:
As written above, the templating is needed because `T` can be 4 different types at a minimum.